tests/unit_tests/dft/include/test_common.hpp

```diff
@@ -0,0 +1,228 @@
+template <typename fp>
+static fp rand_scalar() {
+    return fp(std::rand()) / fp(RAND_MAX) - fp(0.5);
+}
+template <typename fp>
+static std::complex<fp> rand_complex_scalar() {
+    return std::complex<fp>(rand_scalar<fp>(), rand_scalar<fp>());
+}
+template <>
+std::complex<float> rand_scalar() {
+    return rand_complex_scalar<float>();
+}
+template <>
+std::complex<double> rand_scalar() {
+    return rand_complex_scalar<double>();
+}
+
+template <>
+int32_t rand_scalar() {
+    return std::rand() % 256 - 128;
+}
+
+template <typename fp>
+void rand_vector(fp *v, int n, int inc) {
+    int abs_inc = std::abs(inc);
+    for (int i = 0; i < n; i++) {
+        v[i * abs_inc] = rand_scalar<fp>();
+    }
+}
+
+template <typename vec>
+void rand_vector(vec &v, int n, int inc) {
+    using fp = typename vec::value_type;
+    int abs_inc = std::abs(inc);
+
+    v.resize(n * abs_inc);
+
+    for (int i = 0; i < n; i++) {
+        v[i * abs_inc] = rand_scalar<fp>();
+    }
+}
+
+// Catch asynchronous exceptions.
+auto exception_handler = [](sycl::exception_list exceptions) {
+    for (std::exception_ptr const &e : exceptions) {
+        try {
+            std::rethrow_exception(e);
+        }
+        catch (sycl::exception e) {
+            std::cout << "Caught asynchronous SYCL exception:\n" << e.what() << std::endl;
+            print_error_code(e);
+        }
+    }
+};
+
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+void commit_descriptor(oneapi::mkl::dft::descriptor<precision, domain> &descriptor,
+                       sycl::queue queue) {
+#ifdef CALL_RT_API
+    descriptor.commit(queue);
+#else
+    TEST_RUN_CT_SELECT_NO_ARGS(queue, descriptor.commit);
+#endif
+}
+
+template <typename T, int D>
+void copy_to_host(sycl::queue sycl_queue, sycl::buffer<T, D> &buffer_in, std::vector<T> &host_out) {
+    sycl_queue.submit([&](sycl::handler &cgh) {
+        sycl::accessor accessor = buffer_in.get_access(cgh);
+        cgh.copy(accessor, host_out.data());
+    });
+    sycl_queue.wait();
+}
+
+template <typename T, int D>
+void copy_to_device(sycl::queue sycl_queue, std::vector<T> &host_in,
+                    sycl::buffer<T, D> &buffer_out) {
+    sycl_queue.submit([&](sycl::handler &cgh) {
+        sycl::accessor accessor = buffer_out.get_access(cgh);
+        cgh.copy(host_in.data(), accessor);
+    });
+    sycl_queue.wait();
+}
+
+class DimensionsDeviceNamePrint {
+public:
+    std::string operator()(
+        testing::TestParamInfo<std::tuple<sycl::device *, std::int64_t>> dev) const {
+        std::string size = "size_" + std::to_string(std::get<1>(dev.param));
+        std::string dev_name = std::get<0>(dev.param)->get_info<sycl::info::device::name>();
+        for (std::string::size_type i = 0; i < dev_name.size(); ++i) {
+            if (!isalnum(dev_name[i]))
+                dev_name[i] = '_';
+        }
+        std::string info_name = (size.append("_")).append(dev_name);
```

Comment:
`append` changes the original string and returns a reference, so you are making a copy here for no reason.

Reply:
I've changed this to `std::string info_name = std::move(size) + "_" + std::move(dev_name)` in [56d18e6](https://github.com/oneapi-src/oneMKL/pull/261/commits/56d18e60d45dc3a0b617eef414eb0e84cbe6d7c3).

Reply:
Sorry, I think this is a little awkward to read IMO. I had in mine either directly returning `size` i.e.
```
//...
  }
  size.append("_").append(dev_name);
  return size;
  }
```
or renaming the reference
```
//...
  }
  std::string& info_name = size.append("_").append(dev_name);
  return info_name;
}

Reply:
Yes, it was awkward to read. Changed to your first suggestion in [a306d18](https://github.com/oneapi-src/oneMKL/pull/261/commits/a306d1886f3dcaf4252103de5fa5f0990d126474).